### PR TITLE
[Merged by Bors] - small and mostly pointless refactoring

### PR DESCRIPTION
**Pull request by danieleades**
What is says on the tin.

This has got more to do with making `clippy` slightly more *quiet* than it does with changing anything that might greatly impact readability or performance.

that said, deriving `Default` for a couple of structs is a nice easy win

### CODE REVIEW

**Comment by mockersf** [comment]
> > a lot of those changes are in bevy_render, and will be lost with the current rewrite
> 
> what's the action here? should we wait for that and rebase? or are you just warning me that all my hard work will be for naught?

Not sure, I think `bevy_render` will be replaced by `bevy_render2` which you can find in branch [`pipelined-rendering`](https://github.com/bevyengine/bevy/tree/pipelined-rendering/pipelined). I don't know how much of what you did can be ported, I am not following the rewrite very closely

**Comment by TheRawMeatball** [comment]
Does such a thing as a Clippy.toml exist? From a quick search, these seem to not quite exist: https://github.com/rust-lang/cargo/issues/5034, which is why I didn't suggest it.

**Comment by IceSentry** [comment]
Adding this to `.cargo/config.toml` seems to work

```toml
[target.'cfg(feature = "cargo-clippy")']
rustflags = ["-Aclippy::type_complexity"]
```

But that would required adding config.toml to the repository and I'm not sure if we want that, but it could at least be added to the recommended configs.

**Comment by mockersf** [comment]
New renderer has been merged, causing a lot of conflicts. Do you want to update this PR?

**Comment by danieleades** [comment]
turned out to be easier to simply start again...

question for the maintainers- are you happy with me simply fixing these lints, or would you like me to also add compiler flags so that they *stay* fixed?

**Comment by danieleades** [comment]
that's probably enough changes for one PR. I'll reserve further changes for future PRs

File: crates/bevy_tasks/src/iter/adapters.rs
Comment (danieleades) [inline crates/bevy_tasks/src/iter/adapters.rs:201]: was this bound meant to be `Clone` by any chance?

**Comment by danieleades** [comment]
rebased. any interest in merging this?

**Comment by alice-i-cecile** [comment]
> rebased. any interest in merging this?

Yeah, this seems nice. Let me do a quick review.

File: crates/bevy_tasks/src/iter/adapters.rs
Comment (alice-i-cecile) [reply crates/bevy_tasks/src/iter/adapters.rs:201]: Yeah, that seems suspicious.

**Review by alice-i-cecile — APPROVED** [review]
Nothing controversial in here, and these are good tiny code quality fixes. You have my approval, although we should verify the open comment on this PR before merging.

File: crates/bevy_tasks/src/iter/adapters.rs
Comment (cart) [reply crates/bevy_tasks/src/iter/adapters.rs:183]: Probably meant to be Clone. But lets revert this for now. I don't want to conflate functionality changes / bugfixes with style changes.

Comment (cart) [reply crates/bevy_tasks/src/iter/adapters.rs:183]: (feel free to open a new pr or issue to discuss and address this)

**Comment by cart** [comment]
Looks good to me! Resolve that last conflict and revert the `iter::Copied` change and we're good to go!

**Review by alice-i-cecile — APPROVED** [review]
@cart I've done a fresh review pass and the changes you requested are complete :) This PR LGTM now.

**Pull request closed without merging**
